2_standard_libraries/collections_examples.py

```diff
@@ -0,0 +1,117 @@
+from collections import ChainMap, Counter, namedtuple, deque, OrderedDict, defaultdict
+
+def do_chainmap():
+    # Chainmap - linking objects, faster than dictionary or update(calls)
+
+    dict1 = {'x':1, 'y':2,'z':3,'a':4,'b':5}
+    dict2 = {1:'a', 2:'b'}
+    my_map = list(ChainMap(dict1, dict2))
+
+    dict3 = {1:222, 2:333, 3:444}
+    dict4 = {1: 555, 2: 666, 3: 777}
+    c = ChainMap(dict3)
+    d = c.new_child(dict4) # contains dict3 from parent
+    mapped_second_dict_in_d = d.maps[1]
+    c['val'] = 2 # set value for 'val' in context c
+    del c[3] # removed 3:444 from dict3
+
+def do_some_counting():
+    cnt = Counter('abc!')
+    cnt2 = Counter({'a': 2, 'b': 3, 'c': 2})
+    elems = sorted(cnt2.elements())
+    mst_common = cnt2.most_common(1)  # first most common elem
+    cnt.subtract(cnt2)  # subtract cnt2 from cnt
+    total = cnt2.total()
+
+    a = Counter(a=1, b=2, c=3)
+    b = Counter(a=0, b=1, c=3)
+    subtr = a - b  # math operations are possible with counter objects
+    union_max = a | b
+    c = Counter(a=-2, b=1, c=0)
+    removed_zero_and_negative = +c
+    c.update(a=2, c=-2, d=8)
+
+def do_named_tuple():
+    '''named tuples for data assigning'''
+    Cat = namedtuple('Cat', 'size sex color')
+    my_cat = Cat(13, 'male', 'black')
+    size, sex, color = my_cat  # unpacking
+
+    vector = namedtuple('Vector', ['pointX', 'pointY'])
```

Comment:
usually, when we use `namedtuple`, the variable for it should be capitalized: `Vector` so that it is read not as a regular variable but as a class or something that needs attention.